solutions/azure-automation-state-configuration/azuredeploy.bicep

```diff
@@ -0,0 +1,594 @@
+param location string = resourceGroup().location
+
+param adminUserName string
+
+@secure()
+param adminPassword string
+param emailAddress string
+param windowsVMCount int = 1
+param linuxVMCount int = 1
+param vmSize string = 'Standard_DS1_v2'
+param windowsConfiguration object = {
+  name: 'windowsfeatures'
+  description: 'A configuration for installing IIS.'
+  script: 'https://raw.githubusercontent.com/mspnp/samples/main/solutions/azure-automation-state-configuration/scripts/windows-config.ps1'
+}
+param linuxConfiguration object = {
+  name: 'linuxpackage'
+  description: 'A configuration for installing Nginx.'
+  script: 'https://raw.githubusercontent.com/mspnp/samples/main/solutions/azure-automation-state-configuration/scripts/linux-config.ps1'
+}
+param virtualNetworkName string = 'virtial-network'
+param addressPrefix string = '10.0.0.0/16'
+param subnetPrefix string = '10.0.0.0/24'
+param subnetName string = 'subnet'
+
+var logAnalyticsName = uniqueString(resourceGroup().id)
+var automationAccountName = uniqueString(resourceGroup().id)
+var moduleUri = 'https://devopsgallerystorage.blob.core.windows.net/packages/nx.1.0.0.nupkg'
+var subnetRef = virtualNetworkName_subnet.id
+var alertQuery = 'AzureDiagnostics\n| where Category == "DscNodeStatus"\n| where ResultType == "Failed"'
+var windowsNicName = 'windows-nic-'
+var windowsPIPName = 'windows-pip-'
+var windowsVMName = 'windows-vm-'
+var windowsOSVersion = '2016-Datacenter'
+var linuxNicName = 'linux-nic-'
+var linuxPIPName = 'linux-pip-'
+var linuxVMNAme = 'linux-vm-'
+var osVersion = '16.04.0-LTS'
+
+resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
+  name: logAnalyticsName
+  location: location
+  properties: {
+    sku: {
+      name: 'PerGB2018'
+    }
+    features: {
+      searchVersion: 1
+    }
+  }
+}
+
+resource logAnalytics_savedSearches 'Microsoft.OperationalInsights/workspaces/savedSearches@2023-09-01' = {
+  parent: logAnalytics
+  name: '${logAnalyticsName}_savedSearches'
+  properties: {
+    category: 'event'
+    displayName: 'Non Compliant DSC Node'
+    query: alertQuery
+    version: 2
+  }
+}
+
+resource non_compliant_dsc 'microsoft.insights/scheduledqueryrules@2018-04-16' = {
```

Comment:
please let's update all apis here and in ARM

Reply:
done bicep, I will be working on arm.

Reply:
done